contrib/check-semver.sh

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
+#
+# Checks the public API of crates, exits with non-zero if there are currently
+# changes to the public API not already committed to in the various api/*.txt
+# files.
+
+set -euo pipefail
+
+NIGHTLY=$(cat nightly-version)
+# Our docs have broken intra doc links if all features are not enabled.
+RUSTDOCFLAGS="-Z unstable-options --document-private-items --document-hidden-items --output-format=json --cap-lints=allow"
+CURRENT_COMMIT=$(git rev-parse HEAD)
+MASTER_COMMIT=$(git rev-parse master)
```

Comment:
Note to myself: this does not work... Why? I added the `fetch-depth: 0`

```yml
      - name: "Checkout repo"
        uses: actions/checkout@v4
        with:
          # Number of commits to fetch. 0 indicates all history for all branches and tags.
          fetch-depth: 0 # we need full history for cargo semver-checks
```

Ref: https://github.com/actions/checkout

Reply:
Is it possibly related to the fact it's running inside a forked repo, or something? Not sure... 🤔

Reply:
@storopoli github provides some magic variables that will have git commit hashes for the base and head branches. You want to use those rather than hardcoding `master`, which aside from not working, is not always correct because some PRs are not against master.

Reply:
Done! Thanks :)